### ISSUE TITLE
conflicting license statements

### ISSUE THREAD

**Issue by onlyjob** [ZoneMinder/ZoneMinder#817]
I've noticed that some files have conflicting license statements. For instance the header of file [scripts/ZoneMinder/lib/ZoneMinder/Trigger/Channel.pm](../blob/master/scripts/ZoneMinder/lib/ZoneMinder/Trigger/Channel.pm) claims GPL-2+ but in the end of the same file I found [the following](../blob/master/scripts/ZoneMinder/lib/ZoneMinder/Trigger/Channel.pm#L161) statement:

> This library is free software; you can redistribute it and/or modify
it under the same terms as Perl itself, either Perl version 5.8.3 or,
at your option, any later version of Perl 5 you may have available.

Please clarify whether such `.pm` files are double-licensed or if only one particular license applies.

All `.pm` files in `scripts/ZoneMinder/lib/ZoneMinder/Trigger` are affected, some in `scripts/ZoneMinder/lib/ZoneMinder/Control` as well as all `*.pm` and `*.pm.in` in `scripts/ZoneMinder/lib/ZoneMinder`.

**Comment by knnniggett** [comment]
@kylejohnson can you get a hold of Phil and see what he thinks about this one?

**Comment by web2wire** [comment]
I'm happy to put everything on the same licence, probably GPL3 nowadays. I suspect the perl ones just had whatever I found in similar modules from elsewhere.

**Comment by onlyjob** [comment]
So at the moment some Perl components (modules and scripts where Perl license is mentioned) are dual-licensed but we can upgrade to GPL-3+ (version 3 or above), right?

Since everything is already under GPL-2+ I suppose it already includes GPL-3+...

If everyone agrees we can just drop Perl license statement or (in case we choose to preserve dual-licensing) move them towards the beginning of files...

**Comment by SteveGilvarry** [comment]
Review and fix all licenses

**Comment by onlyjob** [comment]
No activity doesn't mean that the problem is gone.

**Comment by 996b** [comment]
Yeah not really knowing much about this trying to learn something

**Comment by SteveGilvarry** [comment]
Thanks for the report (and apologies for the decade-long delay). Opened #4911 to resolve this.

**Clarification: only GPL-2-or-later applies.** ZoneMinder as a whole is GPL-2+ (see `COPYING`), and every affected file already carries the GPL-2+ notice in its header. The trailing POD footer — *"the same terms as Perl itself"* — is leftover `h2xs` module-scaffolding boilerplate, not an intentional dual-license election.

Two things made that clear when auditing the files:

1. In every affected file, the **same copyright holder** appears in both the GPL-2+ header and the Perl-terms footer — including the third-party-contributed `Control/*.pm` modules. Each was created by copying an existing module as a template, inheriting the boilerplate footer that nobody edited.
2. The files were contributed to a GPL-2+ project under a GPL-2+ header, so GPL-2+ is already the governing license. Removing the contradictory line is a clarification, not a relicense — it doesn't change the license anyone relied on.

**The fix:** replace the contradictory POD paragraph with a single-line pointer to the canonical license (the full notice already lives in each file's header, so it isn't duplicated):

```
Licensed under the GNU General Public License v2 or later; see the COPYING
file distributed with ZoneMinder for the full text.
```

Copyright and author attribution lines are left untouched.

**Scope:** a whole-repo scan found 60 `.pm` / `.pm.in` files under `scripts/ZoneMinder/lib/ZoneMinder` containing the boilerplate (all `Trigger/*`, the affected `Control/*`, and the top-level `ZoneMinder/*` modules you listed) and none elsewhere. All 60 held the byte-identical block, so the replacement is uniform across them.
